cogs/score.py

```diff
@@ -42,147 +173,542 @@ async def on_ready(self):
+        - Optional: `from:@userA` AND `to:@userB`
+            -If from and to flags, gives the score given from @userA to @userB. `from:all` and `to:all` can be used.
+            Note that it is possible to give usernames without mention. If a username contains a space, the username and flag must be included in quotes, e.g. "to:user name"
 
+        - Optional: `emojitype:type`
+            -If emojitype flag, gives the score for this emojitype (see below for types)
+            -This is set to emojitype:all by default
+
+        - Optional: `self:bool` (true or false)
+            -If true, self-upvotes are counted
+            -This is set to self:false by default
+
+        - Optional: `before:message_id` AND/OR `after:message_id`
+            -If given, will only count reacts given on messages sent before/after the message_id
+
+        -Available emojitypes for this function:
+            -"all" (All emoji),
+            -"unicode" (All unicode emoji),
+            -"custom" (All custom emoji either in the server or not in the server),
+            -"here" (All custom emojis in the server),
+            -"nothere" (All custom emoji not in the server)
+        """
         conn = sqlite3.connect(self.bot.config.db_path)
         c = conn.cursor()
 
-        c.execute("SELECT Score FROM Members WHERE ID = ?", (m_id, ))
-        score = c.fetchone()
-
-        if not score:
-            score = (0, )
-            c.execute("INSERT INTO Members VALUES (?, ?, ?)", (m_id, nick, 0))
-            conn.commit()
+        try:
+            args_dict = await self._get_converted_args_dict(ctx,
+                                                            args,
+                                                            from_xnor_to=True,
+                                                            member=False,
+                                                            emoji=False)
+        except commands.BadArgument as err:
+            await ctx.send(err)
+            return
 
-        await ctx.send("{} score is {}.".format(nick, score[0]))
+        if args_dict["emojitype"] == "score":
+            await ctx.send("Invalid input: Emojitype flag cannot use "
+                           "type score for this function")
+        # get the WHERE conditions and the values
+        where_str, t = self._where_str_and_values_from_args_dict(args_dict)
+        c.execute((f"SELECT printf('%d. %s', "
+                   f"ROW_NUMBER() OVER (ORDER BY count(*) DESC), "
+                   f"ReactionName), printf('%d %s', count(*), "
+                   f"IIF (count(*)!=1, 'times', 'time')) "
+                   f"FROM Reactions "
+                   f"WHERE {where_str} "
+                   f"GROUP BY ReactionName "), t)
+
+        counts = list(zip(*c.fetchall()))
+        if not counts:
+            await ctx.send(embed=discord.Embed(title="No results found"))
+            return
+        names = counts[0]
+        values = counts[1]
```

Comment:
```py
names, values = counts
```
does this work or is counts longer than length 2?

Reply:
Resolved in 95d666f